encoders/firebase-encoders-json/src/main/java/com/google/firebase/encoders/FieldDescriptor.java

```diff
@@ -43,11 +43,23 @@
 public final class FieldDescriptor {
 
   private final String name;
+  private final boolean required;
+  private final Object defaultValue;
```

Comment:
A few consideration regarding `defaultValue`:
* major: It's use will be limited for "composite" types in the case of `@Encodable`-driven code generation. The issue is that annotations cannot have non-constant values in them, so I would not be able to express something like `@Field(defaultValue = new MyPojo(new SubPojo(), "foo"))` and furthermore annotations cannot contain `null` values, so I cannot even say `@Field(defaultValue =null)`.
* minor: It does not appear to be type-safe, i.e. i can put an `int` inside a fieldDescriptor for `float`
* And last - while may not be too big a deal - it auto-boxes for primitives.

All that said, we have to think more about how(if at all) we allow specifying defaults, and if we don't - using proto3-like defaults might be a good start. wdyt?